packages/core/testing/src/r3_test_bed.ts

```diff
@@ -322,17 +345,21 @@ export class TestBedRender3 implements Injector, TestBed {
    */
   overrideProvider(token: any, provider: {useFactory?: Function, useValue?: any, deps?: any[]}):
       void {
+    const providerDef = provider.useFactory ?
+        {provide: token, useFactory: provider.useFactory, deps: provider.deps || []} :
```

Comment:
```suggestion
        {provide: token, useFactory: provider.useFactory, deps: provider.deps || EMPTY_ARRAY} :
```

I guess `deps` is immutable?